mmv1/templates/terraform/examples/Storage_pool_create.tf.erb

```diff
@@ -0,0 +1,36 @@
+
+resource "google_compute_network" "peering_network" {
+  name = "<%= ctx[:vars]['network_name'] %>"
+}
+
+# Create an IP address
+resource "google_compute_global_address" "private_ip_alloc" {
+  name          = "<%= ctx[:vars]['global_name'] %>"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = google_compute_network.peering_network.id
+}
+
+# Create a private connection
+resource "google_service_networking_connection" "default" {
+  network                 = google_compute_network.peering_network.id
+  service                 = "netapp.servicenetworking.goog"
+  reserved_peering_ranges = [google_compute_global_address.private_ip_alloc.name]
+}
+
+resource "google_netapp_storage_pool" "<%= ctx[:primary_resource_id] %>" {
+  storage_pool_id = "<%= ctx[:vars]['pool_name'] %>"
+  location = "us-central1"
+  service_level = "PREMIUM"
+  capacity_gib = "2048"
+  network = google_compute_network.peering_network.id
+  active_directory      = ""
+  description           = ""
+  global_access_allowed = false
+  kms_config            = ""
+  labels                = {}
+  ldap_enabled          = false
+  psa_range             = ""
```

Comment:
Can we test some values instead of empty ones for these fields? Also we can remove the non-required fields, especially the ones with empty values, from this basic example and have them tested in the update full tests.

Reply:
Hi @shuyama1 ,
all the fields mentioned here are optional only.
for Kms_config and Active directory what values we should pass in automated tests as those are independent resources. which means it's required to be already present before using them here.

Reply:
I'd say we'll need to create the resource in the test similar as the network resource here. I'm not sure what format those fields are expecting, but I find some examples that may be helpful. [TestAccSqlDatabaseInstance_ActiveDirectory](https://github.com/hashicorp/terraform-provider-google-beta/blob/20885c6a8686646cc34060ff7becf4d145b466ae/google-beta/services/sql/resource_sql_database_instance_test.go#L1450C18-L1450C18)
[TestAccAlloydbCluster_CMEKInAutomatedBackupIsUpdatable](https://github.com/hashicorp/terraform-provider-google-beta/blob/20885c6a8686646cc34060ff7becf4d145b466ae/google-beta/services/alloydb/resource_alloydb_cluster_test.go#L520)

Reply:
Got it. Would you be able to tests these fields with real resources locally and confirm that these are working. Thanks!

Reply:
@shuyama1 , I've sent you main.tf having fields populated and state file with resource information in google chat.
Please do let me know if that would suffice.

Reply:
@shuyama1 , I've removed the empty string from basic example as asked.

Reply:
`active_directory` and `kms_config` have dependences on resources that are not yet in Terraform google providers. @G-NamanGupta tested them locally and provided evidence confirming that both fields work as intended.